onnx/backend/test/case/node/rotaryembedding.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+def compute_rotary_embedding(
+    input: np.ndarray,
+    position_ids: np.ndarray,
+    sin_cache: np.ndarray,
+    cos_cache: np.ndarray,
+    interleaved: int = 0,
+    rotary_embedding_dim: int = 0,
+    num_heads: int = 0,
+) -> np.ndarray:
+
+    # First ensure input has shape [batch_size, num_heads, seq_len, head_size]
+    batch_size = input.shape[0]
+    sequence_length = input.shape[1]
+    if len(input.shape) == 3:
+        hidden_size = input.shape[2]
+        assert num_heads != 0
+        head_size = int(hidden_size / num_heads)
+        new_shape = [batch_size, sequence_length, num_heads, head_size]
+        input = np.reshape(input, new_shape)
+    assert len(input.shape) == 4
+    head_size = input.shape[3]
+
+    # Fully or partially perform rotation on input based on rotary_embedding_dim attribute
+    if rotary_embedding_dim == 0:
+        # If rotary_embedding_dim not provided, perform full rotation by using head_size * 2
+        rotary_embedding_dim = cos_cache.shape[1] * 2
```

Comment:
head_size or cos_cache.shape[1]? Better to align code and comment.

Reply:
Also, is it division or multiplication by 2?

Reply:
will leave it as head_size